### PR TITLE
#107: Fixed SLC upload and added version check

### LINKED ISSUE #107: Fix SLC upload in release workflow
Using `$RELEASE_DIR` in an action doesn't work, we probably need to use `${{ env.RELEASE_DIR }}`

https://github.com/exasol/sagemaker-extension/blob/4ff43dfd86bd7f8c42ccf205c4655edd87fd91be/.github/workflows/release_droid_upload_github_release_assets.yml#L61

### PR DESCRIPTION
**Pull request by MarleneKress79789**
fixes #107 
I also added a version check. this is copy pasted from Transformers Extension so does not need in depth review

### CODE REVIEW

File: scripts/check_release_version.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+def get_git_version():
+    repo = Repo()
+    assert not repo.bare
+    tag_strings = [t.name for t in repo.tags]
+    tag_strings.sort(reverse=True)
+
+    latest_tag = tag_strings[0].strip()
+    assert len(latest_tag) > 0
```
Comment (Nicoretti) [inline scripts/check_release_version.py:17]: Rather throw an exception indicating what exactly is wrong, e.g.

```python
if not latest_tag:
    raise Exception("Repository does not have any git tags.")
```

File: scripts/check_release_version.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import re
+from pathlib import Path
+
+from packaging import version
+
+from git import Repo
+import toml
+
+
+def get_git_version():
```
Comment (Nicoretti) [inline scripts/check_release_version.py:10]:
```suggestion
def latest_tag():
```

File: scripts/check_release_version.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    repo = Repo()
+    assert not repo.bare
+    tag_strings = [t.name for t in repo.tags]
+    tag_strings.sort(reverse=True)
+
+    latest_tag = tag_strings[0].strip()
```
Comment (Nicoretti) [inline scripts/check_release_version.py:16]: if tags is empty the code will crash here before the assert is even hit.

File: scripts/check_release_version.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    assert len(latest_tag) > 0
+    return latest_tag
+
+
+def get_poetry_version():
```
Comment (Nicoretti) [inline scripts/check_release_version.py:21]:
```suggestion
def poetry_version():
```

File: scripts/check_release_version.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    parsed_toml = toml.load('pyproject.toml')
+    return parsed_toml["tool"]["poetry"]["version"].strip()
+
+
+def get_change_log_version():
```
Comment (Nicoretti) [inline scripts/check_release_version.py:26]:
```suggestion
def change_log_version():
```

File: scripts/check_release_version.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    # Path overloads __truediv__
```
Comment (Nicoretti) [inline scripts/check_release_version.py:27]: ?

**Review by ckunki — CHANGES_REQUESTED** [review]

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 7]
```diff
@@ -58,4 +58,4 @@ jobs:
         uses: shogo82148/actions-upload-release-asset@v1
         with:
           upload_url: ${{ github.event.inputs.upload_url }}
-          asset_path: $RELEASE_DIR/*
+          asset_path: ${{ env.RELEASE_DIR }}/*
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply .github/workflows/release_droid_upload_github_release_assets.yml:61]: we do that in line 18: https://github.com/exasol/sagemaker-extension/blob/65f4e1b36a46cbcae08dcb3c2e2f91f08948b823/.github/workflows/release_droid_upload_github_release_assets.yml#L19

that should have a scope of being valid in this job only

**Review by Nicoretti — APPROVED** [review]

File: scripts/check_release_version.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import re
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply scripts/check_release_version.py:1]: i will work on using the python toolbox probably in the next kehrwoche. as it is this file is just a copy paste of what we do in other repos until python toolbox is used everywhere. i would like to keep it as we kow it works from the other repos and keeps us from doing mistakes with the release version until we do it in a better way later.

File: .github/workflows/release_droid_upload_github_release_assets.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (ckunki) [reply .github/workflows/release_droid_upload_github_release_assets.yml:61]: :+1:

[the remark this answers is not in the document]
Comment (ckunki) [reply .github/workflows/release_droid_upload_github_release_assets.yml:61]: Looking at line 42 I would think, that `$RELEASE_DIR` should also be fine. 
But let's don't get stuck on this. 
As long as it works it should be fine.

File: scripts/check_release_version.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (ckunki) [reply scripts/check_release_version.py:1]: :+1:

**Review by ckunki — APPROVED** [review]

**Pull request merged**